client/my-sites/marketplace/components/review-item/create-review-item.tsx

```diff
@@ -45,6 +47,18 @@ export function MarketplaceCreateReviewItem( props: MarketplaceCreateReviewItemP
 		setRating( 0 );
 	};
 
+	useEffect( () => {
+		if ( canPublishReview ) {
+			setCannotPublishReviewError( '' );
+		} else {
+			setCannotPublishReviewError(
+				translate(
+					'Only active users can leave a review. Please purchase a new subscription of the product to leave a review.'
+				)
+			);
+		}
+	}, [ canPublishReview ] );
```

Comment:
Non-blocking question: Is this needed? I think using the flag `canPublishReview` below is enough, and we can shave off this `useEffect` and the extra `cannotPublishReviewError` piece of state 😄

Reply:
You are right, I started this differently and then in the end the state was not needed. 
Thanks for pointing.

https://github.com/Automattic/wp-calypso/pull/86430/commits/a91912d6b7bfa0861dae0dc7aff3bb83a4d4776e